### ISSUE TITLE
When starting commit after editing a Shell Script, Commit buffer is in Shell Script mode, not Git-Commit-mode

### ISSUE THREAD

**Issue by jasonm23** [magit/magit#1718]
![screen shot 2015-02-02 at 3 15 50 pm](https://cloud.githubusercontent.com/assets/71587/5996916/68526bc8-aaee-11e4-87be-73441efe69b5.png)

**Comment by tarsius** [comment]
If you are using the `next` branch, check option `git-commit-major-mode`. Otherwise you are on your own I am afraid, I am going on vacation now. Sounds like a misconfiguration (maybe added by some other package?). You might have more luck on the Emacs Stackexchange while I am gone.

**Comment by jasonm23** [comment]
Thanks @tarsius I am coping just fine by switching the mode with my own hook, I'll take a serious look at it later.  Enjoy your vacation!

**Comment by tarsius** [comment]
I've tried again and cannot reproduce this.

**Comment by tarsius** [comment]
Ask if you need help investigating, but since I cannot reproduce it likely is a local problem putting you in a much better position to figure out what's going on.

**Comment by jasonm23** [comment]
Thanks for looking, I'll try to isolate the problem

**Comment by jasonm23** [comment]
I finally found the problem, I had a poorly crafted pattern in `auto-mode-alist` ... `\\.zsh` so anything which included `.zsh` in the name regardless of position, triggered the thing.

Just coincidental that the scripts I edit most often are in a folder called `.zsh.d` ... so it didn't even matter which file was being edited in there.

Anyway, I updated the pattern to be `\\.zsh$` instead.  Problem solved.
